src/java.base/share/classes/java/lang/template/StringTemplate.java

```diff
@@ -0,0 +1,368 @@
+ * ...
+ * String result = TemplateRuntime.interpolate(fragments, values);;
+ * }
+ *
+ * @implSpec An instance of {@link StringTemplate} is immutatble. Also, the
+ * fragment list size must be one more than the values list size.
+ *
+ * @see ValidatingProcessor
+ * @see TemplateProcessor
+ * @see StringProcessor
+ * @see java.util.FormatProcessor
+ *
+ * @since 20
+ */
+@PreviewFeature(feature=PreviewFeature.Feature.STRING_TEMPLATES)
+public interface StringTemplate {
+    /**
+     * Returns an immutable list of string fragments consisting of the string
+     * of characters preceeding each of the embedded expressions plus the
+     * string of characters following the last embedded expression. In the
+     * example: {@snippet :
+     * StringTemplate st = RAW."The student \{student} is in \{teacher}'s class room.";
+     * List<String> fragments = st.fragments(); // @highlight substring="fragments()"
+     * }
+     * <code>fragments</code> will be equivalent to
+     * <code>List.of("The student ", " is in ", "'s class room.")</code>
+     *
+     * @return list of string fragments
+     *
+     * @implSpec The list returned is immutable.
+     */
+    List<String> fragments();
+
+    /**
+     * Returns an immutable list of embedded expression results. In the example:
+     * {@snippet :
+     * StringTemplate st = RAW."\{x} + \{y} = \{x + y}";
+     * List<Object> values = st.values(); // @highlight substring="values()"
+     * }
+     * <code>values</code> will be equivalent to <code>List.of(x, y, x + y)</code>
+     *
+     * @return list of expression values
+     *
+     * @implSpec The list returned is immutable.
+     */
+    List<Object> values();
+
+    /**
+     * {@return the interpolation of the StringTemplate}
+     */
+    default String interpolate() {
+        return TemplateRuntime.interpolate(fragments(), values());
+    }
+
+    /**
+     * Returns the result of applying the specified processor to this {@link StringTemplate}.
+     * This method can be used as an alternative to string template expressions. For example,
+     * {@snippet :
+     * String result1 = STR."\{x} + \{y} = \{x + y}";
+     * String result2 = RAW."\{x} + \{y} = \{x + y}".process(STR); // @highlight substring="process"
+     * }
+     * produces an equivalent result for both {@code result1} and {@code result2}.
+     *
+     * @param processor the {@link ValidatingProcessor} instance to process
+     *
+     * @param <R>  Processor's process result type.
+     * @param <E>  Exception thrown type.
+     *
+     * @return constructed object of type R
+     *
+     * @throws E exception thrown by the template processor when validation fails
+     * @throws NullPointerException if processor is null
+     *
+     * @implNote The default implementation simply invokes the processor's process
+     * method {@code processor.process(this)}.
+     */
+    default <R, E extends Throwable> R process(ValidatingProcessor<R, E> processor) throws E {
```

Comment:
signature should be `ValidatingProcessor<? extends R, ? extends E> processor`

Reply:
Changing

Reply:
Raw list?

Reply:
I think this comment is not on the right part of the code.